protocol/contracts/libraries/LibUnripe.sol

```diff
@@ -235,6 +239,6 @@ library LibUnripe {
         uint256 supply
     ) internal view returns (uint256 redeem) {
         AppStorage storage s = LibAppStorage.diamondStorage();
-        redeem = s.u[unripeToken].balanceOfUnderlying.mul(amount).div(supply);
+        redeem = (s.u[unripeToken].balanceOfUnderlying ** 2).mul(amount).div(supply ** 2);
```

Comment:
Have you done any experimentation on whether `(s.u[unripeToken].balanceOfUnderlying ** 2).mul(amount).div(supply ** 2)` vs `(s.u[unripeToken].balanceOfUnderlying ** 2).div(supply ** 2).mul(amount)` is better? At first glance there'd be a larger chance of overflow given the squared value is then multipled. Ideally some analysis should be done to get the bounds of the implmentation. This can be done on a simple foundry test to get a range in which this would overflow.